src/sdk/data-api.ts

```diff
@@ -80,21 +80,28 @@ export class DataApiImpl implements DataApi {
         recordCreate.type,
         recordCreate.fields
       );
+      this.validate_response(response);
       return { id: response.id };
     });
   }
 
   async query(soql: string): Promise<RecordQueryResult> {
     return this.promisifyRequests(async (conn: Connection) => {
-      const response = await conn.query(soql);
-      const records = response.records.map(createCaseInsensitiveRecord);
-
-      return {
-        done: response.done,
-        totalSize: response.totalSize,
-        records,
-        nextRecordsUrl: response.nextRecordsUrl,
-      };
+      try {
+        const response = await conn.query(soql);
+        this.validate_response(response);
+        const records = response.records.map(createCaseInsensitiveRecord);
+        return {
+          done: response.done,
+          totalSize: response.totalSize,
+          records,
+          nextRecordsUrl: response.nextRecordsUrl,
+        };
+      } catch (e) {
+        console.log(e);
+        this.handle_bad_response(e);
+      }
+      return undefined;
```

Comment:
Returning undefined here to prevent against `error TS7030: Not all code paths return a value.` 
Let me know if there's a better way of handling this or if I should just disable `noImplicitReturns` in `tsconfig.json` entirely.

Reply:
I think you could move the `return { done, totalSize, records, nextRecordsUrl }` to below the `try ... catch` block, then you shouldn't need the `return undefined`.

Reply:
I dont think that would work, because response would be undefined and none of those values (done, totalSize, records, nextRecordsUrl) would have been set.